### DISCUSSION TITLE
Is there a way to update the attributes of custom document extension ?

### DISCUSSION THREAD

**Discussion by vikasvk7** [ueberdosis/tiptap#3972] [General Questions]
Hi, I am beginner to tiptap and wanted to know how we can update the attributes of a custom extension of Document Node.

So, currently it looks like this.
```
`{
  "type": "doc",
  "content": [
    {
      "type": "paragraph",
      "content": [
        {
          "type": "text",
          "text": "Wow, this editor instance exports its content as JSON."
        }
      ]
    }
  ]
}`
```

Now I added a custom Extension, that is DocExtension which has an attribute `mentionsPopup` and now the JSON looks like this.

```
`{
  "type": "doc",
  "attrs" : {
        "mentionsPopup" : false,
   },
  "content": [
    {
      "type": "paragraph",
      "content": [
        {
          "type": "text",
          "text": "Wow, this editor instance exports its content as JSON."
        }
      ]
    }
  ]
}`
```

I want to update the "mentionsPopup" value to true or false depending upon some use case (like if there are `mention` nodes in the editor)

I am able to update the attributes of any other nodes which are not of `doc` type through updateAttributes: 
`editor.commands.updateAttributes("heading", { level: 3 });`

But the below code doesn't work : 

`editor.commands.updateAttributes("doc", { mentionsPopup: true });`

Also tried by updating the state, but in react it's giving some error which says, `Attrs` is a read only property, it can't be assigned 

```typescript
  const updateDocNodeAttributes = () => {
    const newState = editor.state;
    newState.doc.attrs["mentionsPopup"] = true;
    editor.view.updateState(newState);
    // empty dispatch to trigger update
    editor.view.dispatch(editor.view.state.tr.setNodeMarkup(0, undefined, {}));
  };
``` 
So, is there a way we can update that attribute ? 
Thanks in advance!

**Comment by palmcivet** [comment]
Hi, I have the problem similar to yours, I customized a Node and used `updateAttributes()` in the command to update the Node, but it doesn't work. After a series of checks, I found that I forgot to add `run()` when calling my custom command, maybe you can also check it。

In addition, I think your `updateDocNodeAttributes()` function needs to return a boolean value, refer to this https://tiptap.dev/api/commands#dry-run-for-commands

```js
// custom extension like this
Node.create({
  addCommands() {
    xxxx: ({ yy }) => {
      return ({ commands }) => {
        return commands.updateAttributes('xx', { yy });
      };
    },
  }
})

// call command like this
editor.chain().focus().deleteFile().run();
```
